batch/batch/worker/worker.py

```diff
@@ -2483,11 +2482,23 @@ async def create_and_start(
         assert os.path.isdir(root_dir)
 
         assert instance_config
-        total_memory_bytes = n_cores * worker_memory_per_core_bytes(CLOUD, instance_config.worker_type())
+
+        if CLOUD == 'gcp':
+            machine_type_parts = gcp_machine_type_to_parts(INSTANCE_CONFIG["machine_type"])
+            assert machine_type_parts
+            total_memory_bytes = gcp_cores_mcpu_to_memory_bytes(
+                n_cores * 1000, machine_type_parts.machine_family, machine_type_parts.worker_type
+            )
+
+        else:
+            assert CLOUD == 'azure'
+            machine_type_parts = azure_machine_type_to_parts(INSTANCE_CONFIG["machine_type"])
+            assert machine_type_parts
+            total_memory_bytes = azure_cores_mcpu_to_memory_bytes(n_cores * 1000, machine_type_parts.family)
 
         # We allocate 60% of memory per core to off heap memory
-        memory_per_core_mib = worker_memory_per_core_mib(CLOUD, instance_config.worker_type())
-        memory_mib = n_cores * memory_per_core_mib
+
+        memory_mib = total_memory_bytes / (1024**2)
```

Comment:
```suggestion
        memory_mib = total_memory_bytes // (1024**2)
```
This needs to result in an integer, not a float